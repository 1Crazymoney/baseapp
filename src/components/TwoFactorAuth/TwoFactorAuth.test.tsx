import * as React from 'react';
import { TwoFactorAuth, TwoFactorAuthProps } from './';

import { shallow } from 'enzyme';

const defaults: TwoFactorAuthProps = {
    onSubmit: jest.fn(),
    onSignUp: jest.fn(),
    signInLabel: '',
    codeLabel: '',
    buttonLabel: '',
    footerCreateAccountLabel: '',
    signUpLabel: '',
    otpCode: '',
    error: '',
    handleOtpCodeChange: jest.fn(),
};

const setup = (props: Partial<TwoFactorAuthProps> = {}) =>
    shallow(<TwoFactorAuth {...{ ...defaults, ...props }} />);

describe('TwoFactorAuth component', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
    });
});
