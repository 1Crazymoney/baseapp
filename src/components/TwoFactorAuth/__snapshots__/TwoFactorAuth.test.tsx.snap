// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoFactorAuth component should render 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="cr-sign-in-form"
  >
    <h1
      className="cr-sign-in-form__title"
      style={
        Object {
          "marginTop": 119,
        }
      }
    />
    <div
      className="cr-sign-in-form__form-content"
    >
      <div
        className="cr-sign-in-form__group"
      >
        <label
          className="cr-sign-in-form__label"
        />
        <e
          className="cr-sign-in-form__input"
          onChangeValue={[MockFunction]}
          value=""
        />
      </div>
      <div
        className="cr-sign-in-form__button-wrapper cr-sign-in-form__button-wrapper--empty"
      >
        <div
          className="cr-sign-in-form__error-message"
        />
        <div
          className="cr-sign-in-form__loader"
        />
        <Component
          className="cr-sign-in-form__button"
          label=""
          onClick={[Function]}
        />
      </div>
      <div
        className="cr-sign-in-form__footer"
      >
        <p
          className="cr-sign-in-form__footer-create"
        />
        <a
          className="cr-sign-in-form__2fa-bottom-section-text"
          onClick={[MockFunction]}
        />
      </div>
    </div>
  </div>
</form>
`;
